test/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/onboarding/SubscriptionServiceImplSpec.scala

```diff
@@ -344,6 +349,57 @@ class SubscriptionServiceImplSpec extends WordSpec with Matchers with MockFactor
 
     }
 
+    "handing request to register without id and subscribe" must {
+
+      val representeeAnswers  = sample[CompleteRepresenteeAnswers]
+      val sapNumber           = "sapNumber"
+      val cgtReferenceNumber  = "cgtReference"
+      val registrationDetails = RegistrationDetails(
+        representeeAnswers.name,
+        representeeAnswers.contactDetails.emailAddress,
+        representeeAnswers.contactDetails.address,
+        ManuallyEnteredEmail
+      )
+      val subscriptionDetails = SubscriptionDetails(
+        Right(representeeAnswers.name),
+        representeeAnswers.contactDetails.emailAddress,
+        representeeAnswers.contactDetails.address,
+        representeeAnswers.contactDetails.contactName,
+        SapNumber(sapNumber),
+        ManuallyEnteredEmail,
+        ManuallyEnteredAddress,
+        ManuallyEnteredContactName
+      )
+
+      "user is successfully registered and subscribed returning correct cgt reference" in {
```

Comment:
it'd be good to get some unit tests to cover when:
- register without id fails
- subscription fails
- subscription comes back with an "already subscribed" response

Also, this doesn't sound quite right:
```
SubscriptionServiceImpl when handing request to register without
id and subscribe must user is successfully registered and subscribed 
returning correct cgt reference
````